### PR TITLE
Changes for s390x + added python3-ovirt-engine rpm for s390x

### PR DESCRIPTION
**Pull request by dpkulkarni**
This PR contains the changes needed to build the docker images for s390x arch. 
`python3-ovirt-engine-sdk4-4.4.10-1.el8.s390x.rpm` is not available for s390x with community so the rpm is built locally and added to the build steps to fetch it locally while building. 

 `@miq-bot add-label enhancement`

### CODE REVIEW

File: images/manageiq-base/Dockerfile [hunk 1]
```diff
@@ -31,14 +31,16 @@ LABEL name="manageiq-base" \
...
+RUN if [ ${ARCH} = "s390x" ]; then dnf -y install /tmp/rpms/python3-ovirt-engine-sdk4-4.4.10-1.el8.s390x.rpm ; fi
 COPY container-assets/create_local_yum_repo.sh /
 
 RUN curl -L https://releases.ansible.com/ansible-runner/ansible-runner.el8.repo > /etc/yum.repos.d/ansible-runner.repo
 
-RUN dnf -y --disableplugin=subscription-manager install \
-      http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-linux-repos-8-2.el8.noarch.rpm \
-      http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm \
-      https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm \
+RUN if [ ${ARCH} != "s390x" ] ; then dnf -y install http://mirror.centos.org/centos/8.2.2004/BaseOS/${ARCH}/os/Packages/centos-repos-8.2-2.2004.0.1.el8.${ARCH}.rpm \
```
Comment (bdunne) [inline images/manageiq-base/Dockerfile:40]: I get a 404 on this URL.

File: images/manageiq-base/Dockerfile [hunk 2]
```diff
@@ -31,14 +31,16 @@ LABEL name="manageiq-base" \
       io.openshift.tags="ManageIQ,miq,manageiq"
 
 COPY rpms/* /tmp/rpms/
+COPY s390x/*.rpm /tmp/rpms/
+RUN if [ ${ARCH} = "s390x" ]; then dnf -y install /tmp/rpms/python3-ovirt-engine-sdk4-4.4.10-1.el8.s390x.rpm ; fi
```
Comment (bdunne) [inline images/manageiq-base/Dockerfile:35]: Shouldn't this be included in the repo created on the next line?

**Comment by Fryguy** [comment]
I don't think we should be committing the rpms into this repo.  We should store them in a server somewhere, then leverage the rpms dir that already exists: https://github.com/ManageIQ/manageiq-pods/pull/726/files#diff-4f3e6309fc69d51e15b61e092f967cf80499bd63384f4e7d2da8f24ae5c1bd49R33

@bdunne If we do that, is anything else needed?  That is, won't the system automatically install from that rpms dir if it finds it there?

File: images/manageiq-base/Dockerfile (hunk 1 above)
Comment (dpkulkarni) [reply images/manageiq-base/Dockerfile]: Fixed it

File: images/manageiq-base/Dockerfile [hunk 3]
```diff
@@ -31,13 +31,17 @@ LABEL name="manageiq-base" \
...
 COPY container-assets/create_local_yum_repo.sh /
 
 RUN curl -L https://releases.ansible.com/ansible-runner/ansible-runner.el8.repo > /etc/yum.repos.d/ansible-runner.repo
 
-RUN dnf -y --disableplugin=subscription-manager install \
+RUN if [ ${ARCH} != "s390x" ] ; then dnf -y --disableplugin=subscription-manager install \
       http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-linux-repos-8-2.el8.noarch.rpm \
-      http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm \
+      http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm \ 
+      https://resources.ovirt.org/pub/yum-repo/ovirt-release44.rpm; fi && \
```
Comment (bdunne) [inline images/manageiq-base/Dockerfile:43]: This line already exists a few lines down, so we shouldn't need it inside the conditional as well, right?

**Comment by Fryguy** [comment]
This PR seems to have deleted the images/manageiq-base/rpms/.gitkeep file, but we should leave that in place.

**Review by Fryguy — CHANGES_REQUESTED** [review]
Some concerns as mentioned above

**Comment by dpkulkarni** [comment]
On hold till we have resolved the `rpm` location to be picked up during Z builds

**Comment by dpkulkarni** [comment]
Rebased and removed building `python3-ovirt-engine` rpm for Z @Fryguy @bdunne

File: images/manageiq-base/Dockerfile [hunk 4]
```diff
@@ -35,9 +35,10 @@ COPY container-assets/create_local_yum_repo.sh /
 
 RUN curl -L https://releases.ansible.com/ansible-runner/ansible-runner.el8.repo > /etc/yum.repos.d/ansible-runner.repo
 
-RUN dnf -y --disableplugin=subscription-manager install \
-      http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-stream-repos-8-2.el8.noarch.rpm \
-      http://mirror.centos.org/centos/8-stream/BaseOS/${ARCH}/os/Packages/centos-gpg-keys-8-2.el8.noarch.rpm \
+RUN if [ ${ARCH} != "s390x" ] ; then dnf -y --disableplugin=subscription-manager install \
+      http://mirror.centos.org/centos/8.3.2011/BaseOS/${ARCH}/os/Packages/centos-linux-repos-8-2.el8.noarch.rpm \
```
Comment (bdunne) [inline images/manageiq-base/Dockerfile:39]: These shouldn't be reverted

Comment (dpkulkarni) [reply images/manageiq-base/Dockerfile:39]: These repos does not work for s390x so added a check for non-s390x arch

Comment (Fryguy) [reply images/manageiq-base/Dockerfile:39]: The conditional is fine, but you reverted it from centos stream to centos 8.3

**Comment by Fryguy** [comment]
@dpkulkarni Can you please also put back the images/manageiq-base/rpms/.gitkeep which was deleted?

**Review by bdunne — CHANGES_REQUESTED** [review]
Changes to the `Dockerfile` look good now, just need to remove the `.gitkeep` deletion.

**Review by bdunne — APPROVED** [review]

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 1650d2028d14c50710b505e1ee65286d5ef4c4e8.

```text
commit 1650d2028d14c50710b505e1ee65286d5ef4c4e8
Author: Brandon Dunne <bdunne@redhat.com>
Date:   Tue Jun 29 09:21:21 2021 -0400

    Merge pull request #726 from dpkulkarni/z_changes
    
    Changes for s390x + added python3-ovirt-engine rpm for s390x
    
    (cherry picked from commit ec8aeffa97a02acfdd35cd0620da1b8d78e85d2f)
```

**Pull request merged**